httpbp/response.go

```diff
@@ -0,0 +1,210 @@
+package httpbp
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"html/template"
+	"net/http"
+
+	httpgk "github.com/go-kit/kit/transport/http"
+)
+
+// HTTPError is a specialized error that is returned be the Err method specified
+// in the ErrorResponse interface.
+type HTTPError struct {
+	// Code is the status code to set on the HTTP response.  Defaults to 500 if
+	// it is not set.
+	Code int
+
+	// ResponseMessage is an optional message that can be returned to the
+	// client.  Defaults to the native http.StatusText message for the
+	// StatusCode() of the HTTPError.
+	ResponseMessage string
+
+	// Cause is an optional error that can be used to
+	Cause error
+}
+
+// Error returns the error message to send to the client.
+func (e HTTPError) Error() string {
+	if e.ResponseMessage != "" {
+		return e.ResponseMessage
+	}
+	return http.StatusText(e.StatusCode())
+}
+
+// As implements helper interface for errors.As.
+//
+// If v is pointer to either HTTPError or *HTTPError,
+// *v will be set into this error.
+// Otherwise, As will try errors.As against the Cause of the HTTPError
+func (e HTTPError) As(v interface{}) bool {
+	if target, ok := v.(*HTTPError); ok {
+		*target = e
+		return true
+	} else if target, ok := v.(**HTTPError); ok {
+		*target = &e
+		return true
+	}
+	return errors.As(e.Cause, v)
+}
+
+// Is implements helper interface for errors.Is.
+//
+// Returns true if target is equal to the HTTPError.
+// Otherwise, Is will try errors.Is against the Cause of the HTTPError
+func (e HTTPError) Is(target error) bool {
```

Comment:
`Is` is certainly not useful here. It's only used to compare against simple errors (that comes from `errors.New`). For custom error types `Is` won't work.

Reply:
🔪